driver/ssh/config.go

```diff
@@ -27,14 +28,18 @@ func (conf *Config) EncodeWithDSN(base string) (string, error) {
 	updatedBase := parsed.String()
 	split := strings.Split(updatedBase, "://")
 
+	if len(split) < 2 {
+		return "", errors.New("invalid dsn format to read the warehouse dsn")
```

Comment:
Does the word `warehouse` belong here? Isn't this lib just about SQL tunnels?

```suggestion
		return "", fmt.Errorf("invalid DSN format: splitting by :// gives unexpected results")
```

We could argue that `len(split) != 2` might be more robust also. WDYT?